server/vulnerabilities/macoffice/analyzer.go

```diff
@@ -0,0 +1,187 @@
+package macoffice
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/io"
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/utils"
+)
+
+func latestReleaseNotes(vulnPath string) (ReleaseNotes, error) {
+	fs := io.NewFSClient(vulnPath)
+
+	files, err := fs.MacOfficeReleaseNotes()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(files) == 0 {
+		return nil, nil
+	}
+
+	sort.Slice(files, func(i, j int) bool { return files[j].Before(files[i]) })
+	filePath := filepath.Join(vulnPath, files[0].String())
+
+	payload, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	relNotes := ReleaseNotes{}
+	err = json.Unmarshal(payload, &relNotes)
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensure the release notes are sorted by release date, this is because the vuln. processing
+	// algo. will stop when a release note older than the current software version is found.
+	sort.Slice(relNotes, func(i, j int) bool { return relNotes[j].Date.Before(relNotes[i].Date) })
+
+	return relNotes, nil
+}
+
+func collectVulnerabilities(
+	software *fleet.Software,
+	product ProductType,
+	relNotes ReleaseNotes,
+) []fleet.SoftwareVulnerability {
+	var vulns []fleet.SoftwareVulnerability
+	for _, relNote := range relNotes {
+		// We only care about release notes with set versions and with security updates,
+		// 'relNotes' should only contain valid release notes, but this check not expensive.
+		if !relNote.Valid() {
+			continue
+		}
+
+		if relNote.CmpVersion(software.Version) <= 0 {
+			return vulns
+		}
+
+		for _, cve := range relNote.CollectVulnerabilities(product) {
+			vulns = append(vulns, fleet.SoftwareVulnerability{
+				SoftwareID: software.ID,
+				CVE:        cve,
+			})
+		}
+	}
+	return vulns
+}
+
+func storedVulnerabilities(
+	ctx context.Context,
+	ds fleet.Datastore,
+	softwareID uint,
+) ([]fleet.SoftwareVulnerability, error) {
+	storedSoftware, err := ds.SoftwareByID(ctx, softwareID, false)
+	if err != nil {
+		return nil, err
+	}
+
+	var result []fleet.SoftwareVulnerability
+	for _, v := range storedSoftware.Vulnerabilities {
+		result = append(result, fleet.SoftwareVulnerability{
+			SoftwareID: storedSoftware.ID,
+			CVE:        v.CVE,
+		})
+	}
+	return result, nil
+}
+
+func syncDB(
```

Comment:
Maybe rename this function to `updateVulnsInDB` or something that will describe the fact that we delete old vulns and add new ones?